src/com/fediroryshchuk/homework7/threads/synchronize/semaphor/ReduceCounter.java

```diff
@@ -0,0 +1,34 @@
+package com.fediroryshchuk.homework7.threads.synchronize.semaphor;
+
+import java.util.concurrent.Semaphore;
+
+public class ReduceCounter implements Runnable {
+    private String name;
+    private Semaphore smph;
+
+    ReduceCounter(Semaphore mph, String name) {
+        this.smph = smph;
+        this.name = name;
+        new Thread(this).start();
+    }
+
+    public void run() {
+        System.out.println("Start thread " + name);
+        try {
+            System.out.println("Thread " + name + " are waiting permission");
+            smph.acquire();
+            System.out.println("Thread " + name + " are getting permission");
+
+            for (int i = 0; i < 5; i++) {
+                Count.counter--;
+                System.out.println(name + ": " + Count.counter);
+
+                Thread.sleep(10);
```

Comment:
Reuse your util method

Reply:
I did not quite understand - I should also write a separate method for incrementing the counter